packages/block-library/src/fit-text/index.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Server-side rendering of the `core/fit-text` block.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Renders the `core/fit-text` block on the server.
+ *
+ * @since 22.0.0
+ *
+ * @param array  $attributes Block attributes.
+ * @param string $content    Block default content.
+ *
+ * @return string Returns the block content.
+ */
+function render_block_core_fit_text( $attributes, $content ) {
+	if ( ! $content || is_admin() ) {
+		return $content;
+	}
+
+	// Enqueue the frontend script module.
+	wp_enqueue_script_module( '@wordpress/block-library/fit-text/view' );
+
+	// Add Interactivity API directives for fit text to work with client-side navigation.
+	$processor = new WP_HTML_Tag_Processor( $content );
+	if ( $processor->next_tag() ) {
+		if ( ! $processor->get_attribute( 'data-wp-interactive' ) ) {
+			$processor->set_attribute( 'data-wp-interactive', true );
+		}
+		$processor->set_attribute( 'data-wp-context---core-fit-text', 'core/fit-text::{"fontSize":""}' );
+		$processor->set_attribute( 'data-wp-init---core-fit-text', 'core/fit-text::callbacks.init' );
+		$processor->set_attribute( 'data-wp-style--font-size', 'core/fit-text::context.fontSize' );
```

Comment:
You don't need to use namespaces in the directives now, as this is an interactive block, and it's those who extend this block that should use their own namespaces.


```suggestion
		$processor->set_attribute( 'data-wp-interactive', 'core/fit-text' );
		$processor->set_attribute( 'data-wp-context', '{"fontSize":""}' );
		$processor->set_attribute( 'data-wp-init', 'callbacks.init' );
		$processor->set_attribute( 'data-wp-style', 'context.fontSize' );
```